src/contracts/core/AVSDirectory.sol

```diff
@@ -261,6 +264,27 @@ contract AVSDirectory is
         _deregisterFromOperatorSets(msg.sender, operator, operatorSetIds);
     }
 
+
+    /**
+     * @notice Called by operators to set their allocation delay. 
+     * @param delay the allocation delay in seconds
+     * @dev msg.sender is assumed to be the operator
+     * @dev Fails if setting a delay would result in the ability to set an
+     *      an allocation that is active BEFORE the latest set allocation. 
+     */
+    function setAllocationDelay(uint32 delay) external {
+        require(
```

Comment:
we need to require that the delay is nonzero otherwise it will be set to 21 days